innovation_sweet_spots/pipeline/pilot/timeseries_gtr.py

```diff
@@ -4,43 +4,62 @@
 from innovation_sweet_spots import PROJECT_DIR, logging
 from innovation_sweet_spots.utils.io import import_config
 import innovation_sweet_spots.analysis.analysis_utils as au
+import innovation_sweet_spots.analysis.wrangling_utils as wu
+from innovation_sweet_spots.getters.gtr import get_pilot_GtR_projects
 import pandas as pd
 
-# Input files
-DATA_DIR = PROJECT_DIR / "outputs/finals/pilot_outputs/"
-PROJECTS_TABLE = "ISS_pilot_GtR_projects.csv"
-# Output files
-EXPORT_DIR = DATA_DIR / "time_series/"
-OUTFILE_NAME = "Time_series_GtR_{}.csv"
 # Parameters (tech categories to process, time series limits)
 PARAMS = import_config("iss_pilot.yaml")
+SPLIT = True
+
+# Output files
+EXPORT_DIR = PROJECT_DIR / "outputs/finals/pilot_outputs/time_series/"
+OUTFILE_NAME = "Time_series_GtR_split_{}.csv" if SPLIT else "Time_series_GtR_{}.csv"
 
 if __name__ == "__main__":
 
     EXPORT_DIR.mkdir(parents=True, exist_ok=True)
 
     # Load data
-    gtr_docs = pd.read_csv(DATA_DIR / PROJECTS_TABLE)
+    gtr_docs = au.filter_years(
+        df=get_pilot_GtR_projects(),
+        date_col="start",
+        keep_year_col=False,
+        min_year=PARAMS["min_year"],
+        max_year=PARAMS["max_year"],
+    )
```

Comment:
I have changed this file so that filtering out certain years happens before any processing / splitting of the data. 
There are some cases where for example a project starts in 2019 and ends in 2022. When this project's funding is split there will be records created for 2019, 2020, 2021, 2022.  If the filtering (keep 2007-2021) is done after the splitting, it will filter out the 2022 record so some of the data will be lost.

Filtering the data before processing/splitting means the split time series and non split time series have the same total sums for research funding.

Reply:
Thanks for the explanation!